instance/tests/models/test_server.py

```diff
@@ -337,15 +337,21 @@ def test_vm_not_created(self, server_status):
         ('ready-server-id', ServerStatus.Ready),
     )
     @unpack
-    def test_terminate_server_vm_created(self, openstack_id, server_status):
+    @patch('instance.ssh.remove_known_host_key')
+    def test_terminate_server_vm_created(self, openstack_id, server_status, remove_key):
         """
         Terminate a server with a VM
         """
-        server = OpenStackServerFactory(openstack_id=openstack_id, status=server_status)
+        server = OpenStackServerFactory(
+            openstack_id=openstack_id,
+            status=server_status,
+            # fake a public ip so SSH key can be removed
+            _public_ip='127.0.0.1'
```

Comment:
@kewne, can you add this to the factory object creation code itself instead of repeating it multiple times in similar tests. Afaik, that wouldn't cause any undesirable side effects.

Reply:
@lgp171188 I considered that but decided to do this instead because it's less surprising behavior: if you move it into the factory, code that uses `public_ip` will surprisingly never call through to OS and, to get that behavior, you have to use the even less obvious hack of setting `_public_ip = None`.

Another option is creating a specific Factory for this but that doesn't seem like a good solution either; tbh I think the root issue here is the way `public_ip` is handled.

Reply:
@kewne, my suggestion was to set a public IP as a default when there is none set. So what would be the issue then? I am not sure if I understand this

> code that uses public_ip will surprisingly never call through to OS and, to get that behavior, you have to use the even less obvious hack of setting _public_ip = None.